File: outgoing-simple-java-demo/src/main/java/de/codecentric/cm4sb/demo/outgoing/service/MovieService.java
package de.codecentric.cm4sb.demo.outgoing.service;

import de.codecentric.cm4sb.demo.outgoing.domain.Movie;
import de.codecentric.cm4sb.demo.outgoing.repository.MovieRepository;
import io.vavr.control.Try;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

@Service
public class MovieService {
    private final MovieRepository movieRepository;
    private final Movie fallbackMovie = new Movie("Titanic", "3h 15m");

    public MovieService(MovieRepository movieRepository) {
        this.movieRepository = movieRepository;
    }

    public Movie getMovie() {
        return Try.of(this::getRecommendedMovie).getOrElse(fallbackMovie);
    }

    public Movie getMovieByTitle(String title) {
        return movieRepository.getByTitle(title);
    }

    private List<Movie> getMovies() {
        List<Movie> movies = new ArrayList<>();
        movieRepository.findAll().forEach(movies::add);
        return movies;
    }

    public Movie getRecommendedMovie() {
        List<Movie> movies = getMovies();
        return movies.get(new Random().nextInt(movies.size()));
    }
}

File: outgoing-simple-kotlin-demo/src/main/kotlin/de/codecentric/cm4sb/demo/outgoing/service/MovieService.kt
package de.codecentric.cm4sb.demo.outgoing.service

import de.codecentric.cm4sb.demo.outgoing.domain.Movie
import de.codecentric.cm4sb.demo.outgoing.repository.MovieRepository
import io.vavr.control.Try
import org.springframework.stereotype.Service
import java.util.*

@Service
class MovieService(val movieRepository: MovieRepository) {
    private val fallbackMovie: Movie = Movie("Titanic", "3h 15m")

    fun getMovies(): List<Movie> = movieRepository.findAll().toList()

    fun getMovie(): Movie = Try.of(this::getRecommendedMovie).getOrElse(fallbackMovie)

    fun getMovieByTitle(title: String): Movie = movieRepository.getByTitle(title)

    fun getRecommendedMovie(): Movie {
        val movies = getMovies()
        return movies[Random().nextInt(movies.size)]
    }

}
